### PR TITLE
preserve UID on course home pages when possible

### PR DESCRIPTION
**Pull request by gumaerc**
#### Pre-Flight checklist

- [x] Testing
  - [x] Code is tested
  - [x] Changes have been manually tested

#### What are the relevant tickets?
Closes https://github.com/mitodl/ocw-to-hugo/issues/154

#### What's this PR do?
This PR sets the `uid` and menu `identifier` property on course home pages to the original uid from Plone

#### How should this be manually tested?
Convert any number of courses and assert that each course home page markdown file contains a `uid` property matching the uid of the course home page from the source data.

### CODE REVIEW

**Comment by noisecapella** [comment]
Is this ready for review yet?

**Comment by gumaerc** [comment]
@noisecapella yes, ready for a look now.

File: src/markdown_generators.js [hunk 1]
```diff
@@ -226,6 +228,7 @@ const generateCourseSectionFrontMatter = (
     Generate the front matter metadata for a course section given a title and menu index
     */
   const courseSectionFrontMatter = {
+    uid:       pageId,
```
Comment (noisecapella) [inline src/markdown_generators.js:231]: This is value is `course_home` for `res-2-006-girls-who-build-cameras-summer-2016`. Is that the right value for this?

Comment (gumaerc) [reply src/markdown_generators.js:231]: I'll need to look into this tomorrow, but I think there might be a data issue with some of the courses.  That value is a default in the event the course home page is not found for whatever reason.  Stay tuned.

**Comment by mbertrand** [comment]
I just ran this branch against course `18-01-single-variable-calculus-fall-2006` and the page that previously had no uid (`18-01-single-variable-calculus-fall-2006/sections/readings/course-reader.md`) now does!

```
---
uid: 98fdab81f6f5a25624a5128fe1261e9b
title: Course Reader
course_id: 18-01-single-variable-calculus-fall-2006
type: course
layout: course_section
---
```

File: src/markdown_generators.js (hunk 1 above)
Comment (gumaerc) [reply src/markdown_generators.js:231]: @noisecapella I spent some time looking into this today, and digging through the Plone JSON exports for that course I didn't find any pages of the type `CourseHomeSection`.  It doesn't seem to have a "course home" in the original data, in which case the default of `course_home` makes sense for now I think.

Comment (gumaerc) [reply src/markdown_generators.js:231]: Actually, I just found that for "resource" courses like this, the type seems to be `SRHomePage`.  This type will have to be added to the types parsed by `ocw-data-parser` and added to the `course_pages` array.  We can either merge this as-is for now, or hold off until I can push through an `ocw-data-parser` PR to pull in those other page types.

Comment (gumaerc) [reply src/markdown_generators.js:231]: @noisecapella PR here for that: https://github.com/mitodl/ocw-data-parser/pull/117

**Comment by gumaerc** [comment]
@noisecapella this could use another look

**Comment by noisecapella** [comment]
Can we leave the `uid` blank instead of having a default of `course-home`?

**Comment by gumaerc** [comment]
@noisecapella all set, but @mbertrand should be aware of this

**Comment by gumaerc** [comment]
Whoops, a bit of a slip up there and almost committed the wrong type match for SRHomePage but that's all set now.

**Review by noisecapella — APPROVED** [review]

**Pull request merged**